### DISCUSSION TITLE
Feature Request: add support to hide a issue/PR comment

### DISCUSSION THREAD

**Discussion by sebastiaanspeck** [community/community#140663] [Apps, API and Webhooks]
### Select Topic Area

General

### Body

It would be great if we could hide an issue/PR comment using the GitHub REST API. Right now you can only delete a comment, why not hide it using the API?

**Comment by rdhar** [comment]
Keen to have REST API support for hiding issue/PR comments as part of the existing "[update an issue comment](https://docs.github.com/en/rest/issues/comments?apiVersion=2022-11-28#update-an-issue-comment)" endpoint.

For reference, it'd be handy to choose the reason for hiding, as shown in the UI below.
![image](https://github.com/user-attachments/assets/097cc6a4-3772-409a-8a46-4d869a95ede3)

My particular use-case is for programmatic hiding of outdated bot-comments.

**Comment by sin-ack** [comment]
As a workaround one can currently use the GraphQL API through Octokit:

```graphql
mutation {
  minimizeComment(input: { subjectId: "nodes.id here", classifier: OUTDATED }) {
    minimizedComment { isMinimized }
  }
}
```

You can get the `subjectId` with a query like this:

```graphql
query {
  repository(owner: "foo", name: "bar") {
    pullRequest(number: 1) {
      comments(last: 100) {
        nodes {
          # Use this as subjectId
          id
          author { login }
          viewerCanMinimize
        }
      }
    }
  }
}
```

**Reply by jmezzacappa** [reply]
This works to minimize the comment. However, just a heads up: there's this old lingering bug related to the `classifier` field: https://github.com/orgs/community/discussions/19865
